client/state/selectors/get-sites-for-upcoming-transactions.js

```diff
@@ -0,0 +1,25 @@
+/**
+ * External dependencies
+ */
+import { map, uniq } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import createSelector from 'lib/create-selector';
+import { getSite } from 'state/sites/selectors';
+import { getUpcomingBillingTransactions } from 'state/selectors';
+
+const getSitesForUpcomingTransactions = createSelector(
+	( state ) => {
+		const siteIds = uniq( map( getUpcomingBillingTransactions( state ), 'blog_id' ) )
+			.map( Number );
```

Comment:
Should we filter for unexpected values first? `Number( null )` is 0 and `Number( undefined )` is NaN.